contracts/DandelionOrg.sol

```diff
@@ -0,0 +1,421 @@
+        address[] memory tokenRequestAcceptedDepositTokens,
+        address _timeLockToken,
+        uint256[3] memory _timeLockSettings
+    )
+        internal
+    {
+        Redemptions redemptions = _installRedemptionsApp(_dao);
+        TokenRequest tokenRequest = _installTokenRequestApp(_dao, tokenRequestAcceptedDepositTokens);
+        TimeLock timeLock = _installTimeLockApp(_dao, _timeLockToken, _timeLockSettings);
+
+        _cacheDandelionApps(tokenRequest, redemptions, timeLock);
+
+    }
+
+    function _setupPayrollApp(Kernel _dao, ACL _acl, Finance _finance, Voting _voting, uint256[4] memory _payrollSettings) internal {
+
+        (address denominationToken, IFeed priceFeed, uint64 rateExpiryTime, address employeeManager) = _unwrapPayrollSettings(_payrollSettings);
+        address manager = employeeManager == address(0) ? _voting : employeeManager;
+
+        Payroll payroll = _installPayrollApp(_dao, _finance, denominationToken, priceFeed, rateExpiryTime);
+        _createPayrollPermissions(_acl, payroll, manager, _voting, _voting);
+        _grantCreatePaymentPermission(_acl, _finance, payroll);
+    }
+
+    /* REDEMPTIONS */
+
+    function _installRedemptionsApp(Kernel _dao) internal returns (Redemptions) {
+
+        (TokenManager tokenManager, Vault vault,) = _popBaseAppsCache();
+        Redemptions redemptions = Redemptions(_installNonDefaultApp(_dao, REDEMPTIONS_APP_ID));
+        redemptions.initialize(vault, tokenManager);
+        return redemptions;
+    }
+
+
+    function _createRedemptionsPermissions(
+        ACL _acl,
+        Redemptions _redemptions,
+        TokenManager _tokenManager,
+        Voting _voting,
+        Vault _agentOrVault,
+        address _manager
+    )
+        internal
+    {
+
+        _acl.createPermission(_tokenManager, _redemptions, _redemptions.REDEEM_ROLE(), _manager);
+        _acl.createPermission(_voting, _redemptions, _redemptions.ADD_TOKEN_ROLE(), _manager);
+        _acl.createPermission(_voting, _redemptions, _redemptions.REMOVE_TOKEN_ROLE(), _manager);
+
+    }
+
+    /* TOKEN REQUEST */
+
+    function _installTokenRequestApp(Kernel _dao, address[] memory tokenRequestAcceptedDepositTokens) internal returns (TokenRequest) {
+
+        (TokenManager tokenManager, Vault vault,) = _popBaseAppsCache();
+        TokenRequest tokenRequest = TokenRequest(_installNonDefaultApp(_dao, TOKEN_REQUEST_APP_ID));
+        tokenRequest.initialize(tokenManager, vault, tokenRequestAcceptedDepositTokens);
+        return tokenRequest;
+    }
+
+    function _createTokenRequestPermissions(
+        ACL _acl,
+        TokenRequest _tokenRequest,
+        TokenManager _tokenManager,
+        Voting _voting,
+        address _manager
+    )
+        internal
+    {
+        _acl.createPermission(_tokenManager, _tokenRequest, _tokenRequest.SET_TOKEN_MANAGER_ROLE(), _voting);
+        _acl.createPermission(_tokenManager, _tokenRequest, _tokenRequest.SET_VAULT_ROLE(), _voting);
```

Comment:
Shouldn't the two permissions above be granted to the `voting app` (soon to be Dandelion voting)? Otherwise every account holding tokens could change the token or vault address.